client.go

```diff
@@ -214,10 +231,16 @@ func (c *Client) AddMiddleware(m ClientMiddlewareFunc) *Client {
 func (c *Client) setDefaults() {
 	c.queueDeclareSettings = QueueDeclareSettings{
 		Durable:          true,
-		DeleteWhenUnused: true,
+		DeleteWhenUnused: false,
 		Exclusive:        false,
 		NoWait:           false,
-		Args:             nil,
+		Args: map[string]interface{}{
+			// Ensure the queue is deleted automatically when it's unused for
+			// more than the set time. This is to ensure that messages that
+			// are in flight during a reconnect doesn't get lost (which might
+			// happen when using `DeleteWhenUnused`).
+			"x-expires": 1 * 60 * 1000, // 1 minute.
```

Comment:
If the timeout on the request is less than 1 minute, won't the response be discarded either way after a reconnect if it's been more than the offset from the request? Maybe there should be a way to reset the timer on a request after a reconnect so the timeout will be x interval from when connected/retried? Nothing important for now, maybe just a todo for later, I get that this is to replace `DeleteWhenUnused` to avoid losing messages upon reconnect!

Reply:
This header tells RabbitMQ that it should automatically remove the queue after it's not used for more than 1 minute.

The previously used `DeleteWhenUnused` will automatically delete the queue when there are no consumers, which would happen during a reconnect, which means that any messages in the reply-to queue would be deleted during a reconnect.

> This controls for how long a queue can be unused before it is automatically deleted. Unused means the queue has no consumers, the queue has not been recently redeclared (redeclaring renews the lease) https://www.rabbitmq.com/ttl.html#queue-ttl

Reply:
Yeah I got that, and that's great! What I mean is that if a client/request has 10 second timeout and the connection drops after a send, if it then takes > 10 seconds to reconnect the client will throw away the response, right? Because connected or not, this will happen: 

https://github.com/bombsimon/amqp-rpc/blob/39d6bb5e71e76b10b9b20f21316413f077d65953/client.go#L501

Reply:
Yes that's right, a timeout from the users POV is that it will never take longer than the set timeout to return.